velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <folly/executors/IOThreadPoolExecutor.h>
+#include <iostream>
+#include "AbfsUtil.h"
+#include "folly/io/Cursor.h"
+#include "velox/common/file/File.h"
+
+namespace Azure::Storage::Files::DataLake::Models {
+class PathProperties;
+}
+
+namespace facebook::velox::filesystems::abfs {
+using namespace Azure::Storage::Files::DataLake;
+using namespace Azure::Storage::Files::DataLake::Models;
+
+/*
+ * We are using the DFS (Data Lake Storage) endpoint for Azure Blob File write
+ * operations because the DFS endpoint is designed to be compatible with file
+ * operation semantics, such as `Append` to a file and file `Flush` operations.
+ * The legacy Blob endpoint can only be used for blob level append and flush
+ * operations. When using the Blob endpoint, we would need to manually manage
+ * the creation, appending, and committing of file-related blocks.
+ *
+ * However, the Azurite Simulator does not yet support the DFS endpoint.
+ * (For more information, see https://github.com/Azure/Azurite/issues/553 and
+ * https://github.com/Azure/Azurite/issues/409).
+ * You can find a comparison between DFS and Blob endpoints here:
+ * https://github.com/Azure/Azurite/wiki/ADLS-Gen2-Implementation-Guidance
+ *
+ * To facilitate unit testing of file write scenarios, we define the
+ * IBlobStorageFileClient here, which can be mocked during testing.
+ */
+class IBlobStorageFileClient {
```

Comment:
The definition can go to the test file. We can forward declare here.

Reply:
@majetideepak  thank you! So you mean we can put it in such as `abfsWritefileTest.h` and then let `abfsWriteFile.h` foward the declare to use it ? But i am wondering does it a good way to let business code depends on defination from test file ? Do we already have any implemented code example referenece for same situation (using defination defined in test file from business code file ?)

Reply:
I see your point. `IBlobStorageFileClient` interface is implemented inside both `AbfsWriteFile.cpp` and `AbfsFileSystemTest.cpp`. We have to keep it here then.